pkg/collector/corechecks/net/wlan/wlan_darwin.m

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+#include <Foundation/Foundation.h>
+#include <CoreWLAN/CoreWLAN.h>
+#include <CoreLocation/CoreLocation.h>
+#include "wlan_darwin.h"
+
+
+@interface LocationManager : NSObject <CLLocationManagerDelegate>
+@property (nonatomic, strong) CLLocationManager *locationManager;
+@end
+
+@implementation LocationManager
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        [self setupLocationServices];
+    }
+    return self;
+}
+
+- (void)setupLocationServices {
+    self.locationManager = [[CLLocationManager alloc] init];
+    self.locationManager.delegate = self;
+    
+    if (@available(macOS 11.0, *)) {
+        if (self.locationManager.authorizationStatus == kCLAuthorizationStatusNotDetermined) {
+            [self.locationManager requestWhenInUseAuthorization];
+        }
+    } else {
+        NSLog(@"Location services not available on this OS version");
+    }
+}
+
+#pragma mark - CLLocationManagerDelegate
+
+- (void)locationManagerDidChangeAuthorization:(CLLocationManager *)manager {
+    if (@available(macOS 11.0, *)) {
+        CLAuthorizationStatus status = manager.authorizationStatus;
+        
+        switch (status) {
+            case kCLAuthorizationStatusAuthorizedAlways:
+                NSLog(@"Location services authorized!");
+                [self.locationManager startUpdatingLocation];
+                break;
+                
+            case kCLAuthorizationStatusDenied:
+                NSLog(@"Location services denied by user");
+                break;
+                
+            case kCLAuthorizationStatusRestricted:
+                NSLog(@"Location services restricted");
+                break;
+            default:
+                NSLog(@"Location services status undetermined: %d", status);
+                break;
+        }
+    } else {
+        NSLog(@"Location services not available on this OS version");
+    }
+}
+
+- (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray<CLLocation *> *)locations {
+    CLLocation *location = [locations lastObject];
+    NSLog(@"Location updated: %@", location);
+}
+
+- (void)locationManager:(CLLocationManager *)manager didFailWithError:(NSError *)error {
+    NSLog(@"Location update failed with error: %@", error.localizedDescription);
```

Comment:
Where do these messages end up? Would it make sense to use agent logging facilities instead?

Reply:
It definitely would be better, would it be easy to do that? I guess I could use the C API of the agent? I'm not very familiar with that, any help would be appreciated!

Reply:
I see two ways:
- add logging C logging function in cgo, and call it from Objective C (the agent doesn't have a C API that can be used here)
- accumulate errors in the obj-c code and extract them as needed in the go check code.

Reply:
@vickenty Done! I went with the second approach, I return an `errorMessage` field but I don't propagate it to the WiFiInfo Struct in the Go side.